_base/lang.js

```diff
@@ -307,7 +307,7 @@ define(["./kernel", "../has", "../sniff"], function(dojo, has){
 			// summary:
 			//		Returns true if it is a built-in function or some other kind of
 			//		oddball that *should* report as a function but doesn't
-			return it && !lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
+			return !!it && lang.isFunction(it) && /\{\s*\[native code\]\s*\}/.test(String(it)); // Boolean
```

Comment:
I've never heard that rule before.  The current syntax is terser.

I have a more basic objection though.   The function no longer does what it says.  The summary (emphasis added) is

> 	Returns true if it is a built-in function or some other kind of oddball that *should* report as a function **but doesn't**

Now you've changed it to only report things that do report as functions.  Why is that?

Also, the ticket (https://bugs.dojotoolkit.org/ticket/18244) says something about _toArray(), but you haven't changed anything with toArray().